From 01a4b2b31d5279c90e7c596f9321eb0e9ec38d69 Mon Sep 17 00:00:00 2001
From: Michal Zidek <mzidek@redhat.com>
Date: Fri, 21 Nov 2014 20:06:32 +0100
Subject: [PATCH 19/26] util: sss_get_domain_name regex mismatch not fatal
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Assume name is not FQDN if sss_parse_name fails to
match domain with regular expression.

Fixes:
https://fedorahosted.org/sssd/ticket/2487

Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 src/util/usertools.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/util/usertools.c b/src/util/usertools.c
index 16478998d8936cd2e260c1e53db6b68f1563b0f8..2804953a3e854ddf1a122b389ac1e14c4ff7f865 100644
--- a/src/util/usertools.c
+++ b/src/util/usertools.c
@@ -643,7 +643,13 @@ sss_get_domain_name(TALLOC_CTX *mem_ctx,
     /* check if the name already contains domain part */
     if (dom->names != NULL) {
         ret = sss_parse_name(mem_ctx, dom->names, orig_name, &domain, NULL);
-        if (ret != EOK) {
+        if (ret == ERR_REGEX_NOMATCH) {
+            DEBUG(SSSDBG_TRACE_FUNC,
+                  "sss_parse_name could not parse domain from [%s]. "
+                  "Assuming it is not FQDN.\n", orig_name);
+        } else if (ret != EOK) {
+            DEBUG(SSSDBG_TRACE_FUNC,
+                  "sss_parse_name failed [%d]: %s\n", ret, sss_strerror(ret));
             return NULL;
         }
     }
-- 
2.1.0

